WooCommerce/Classes/POS/ViewModels/PointOfSaleDashboardViewModel.swift

```diff
@@ -106,12 +110,40 @@ final class PointOfSaleDashboardViewModel: ObservableObject {
     func cardPaymentTapped() {
         Task { @MainActor in
             showsCreatingOrderSheet = true
-            let order = try await createTestOrder()
+            let order = try await createTestOrder(amount: formattedOrderTotalPrice ?? "15.00")
             showsCreatingOrderSheet = false
-            let _ = try await cardPresentPaymentService.collectPayment(for: order, using: .bluetooth)
+            try await collectPayment(for: order)
+        }
+    }
+
+    @MainActor
+    private func collectPayment(for order: Order) async throws {
+        let _ = try await cardPresentPaymentService.collectPayment(for: order, using: .bluetooth)
+
+        // TODO: Here we should present something to show the payment was successful or not,
+        // and then clear the screen ready for the next transaction.
+    }
+
+    @MainActor
+    func totalsAppeared() async {
+        // Digging in to the connection viewmodel here is a bit of a shortcut, we should improve this.
+        // TODO: Have our own subscription to the connected readers
+        if cardReaderConnectionViewModel.connectionStatus == .connected {
+            // Once we have finalised order creation/update, this check shouldn't be required.
+            // Instead, we should have whatever code does the order creation/update kick off this connection process.
+            guard let order = try? await createOrUpdateTestOrder() else {
+                return DDLogError("Error creating or updating order")
+            }
 
-            // TODO: Here we should present something to show the payment was successful or not,
-            // and then clear the screen ready for the next transaction.
+            do {
+                // Since this function is called from a `task`, it'll be cancelled automatically if the view is removed.
+                // Proper cancellation isn't implemented yet, so that can lead to some unwanted behaviour, but currently
+                // if you go back during a payment, it will still complete.
+                // TODO: We should consider disabling the `Add More` button when the shopper taps their card.
```

Comment:
I agree that this is the ideal UX, thanks for sharing this in the POS channel maybe we can tag the Android devs for more awareness.